From 9cddcfebd95245687d6d48f6d027e731a39d9b3e Mon Sep 17 00:00:00 2001
From: Giovanni Santini <giovannisantini93@yahoo.it>
Date: Mon, 6 Nov 2017 17:19:24 +0100
Subject: [PATCH] Proper Python checks - Fixing the order used for looking for
 Python versions.   While the intended way was to find a working Python 2
 setup   and fall back on Python 3, the order was swapped.   Restore it to the
 proper one. - Add additional Pylint2 patterns   In this way, ArchLinux one is
 found.

---
 GNUmakefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/GNUmakefile b/GNUmakefile
index 1e5d3a5..a166926 100644
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -128,7 +128,7 @@ endef
 # Python 3, we need to do our best to ensure we have Python 2.
 PYTHON3 = python3 python3.6 python-3.6 python
 PYTHON2 = python2 python2.7 python-2.7 python2.6 python-2.6 python
-ALL_PYTHONS = $(PYTHON3) $(PYTHON2)
+ALL_PYTHONS = $(PYTHON2) $(PYTHON3)
 PYTHON_MAJOR := $(shell python -c "import sys; print sys.version[:1]" 2>/dev/null)
 ifeq ($(PYTHON_MAJOR), 2)
 PYTHON := $(shell which python)
@@ -142,7 +142,7 @@ mycfgdir = $(SYSCFG)/obs/services
 LIST_PY_FILES=git ls-tree --name-only -r HEAD | grep '\.py$$'
 PY_FILES=$(shell $(LIST_PY_FILES))

-ALL_PYLINT2 = pylint-2.7 pylint2.7 pylint
+ALL_PYLINT2 = pylint-2.7 pylint2.7 pylint2 pylint
 ALL_PYLINT3 = pylint-3.4 pylint3.4 pylint-3.5 pylint3.5 pylint-3.6 pylint3.6

 PYLINT2 = $(call first_in_path_opt,$(ALL_PYLINT2))
--
2.16.1